vsintegration/src/FSharp.Editor/Navigation/NavigateToSearchService.fs

```diff
@@ -235,53 +118,100 @@ type internal FSharpNavigateToSearchService
         | PatternMatchKind.CamelCaseSubstring -> FSharpNavigateToMatchKind.CamelCaseSubstring
         | PatternMatchKind.CamelCaseNonContiguousSubstring -> FSharpNavigateToMatchKind.CamelCaseNonContiguousSubstring
         | PatternMatchKind.Fuzzy -> FSharpNavigateToMatchKind.Fuzzy
-        | _ -> FSharpNavigateToMatchKind.Fuzzy
+        | _ -> FSharpNavigateToMatchKind.None
+
+    let createMatcherFor searchPattern =
+        let patternMatcher =
+            lazy
+                patternMatcherFactory.CreatePatternMatcher(
+                    searchPattern,
+                    PatternMatcherCreationOptions(
+                        cultureInfo = CultureInfo.CurrentUICulture,
+                        flags =
+                            (PatternMatcherCreationFlags.AllowFuzzyMatching
+                             ||| PatternMatcherCreationFlags.AllowSimpleSubstringMatching)
+                    )
+                )
+        // PatternMatcher will not match operators and some backtick escaped identifiers.
+        // To handle them, do a simple substring match first.
+        fun (name: string) ->
+            match name.IndexOf(searchPattern, StringComparison.CurrentCultureIgnoreCase) with
+            | i when i > 0 -> PatternMatch(PatternMatchKind.Substring, false, true) |> Some
+            | 0 when name.Length = searchPattern.Length -> PatternMatch(PatternMatchKind.Exact, false, true) |> Some
+            | 0 -> PatternMatch(PatternMatchKind.Prefix, false, true) |> Some
+            | _ -> patternMatcher.Value.TryMatch name |> Option.ofNullable
+
+    let processDocument (tryMatch: string -> PatternMatch option) (kinds: IImmutableSet<string>) (document: Document) =
+        async {
+            let! sourceText = document.GetTextAsync Async.DefaultCancellationToken |> Async.AwaitTask
+
+            let processItem item =
+                asyncMaybe {
+                    do! Option.guard (kinds.Contains(navigateToItemKindToRoslynKind item.Kind))
+
+                    let name =
+                        // This conversion back from logical names to display names should never be needed, because
```

Comment:
So then why is this here? :)

Reply:
You wouldn't believe how old this comment is 😢

Reply:
Well maybe you can remove all this crap for good now since you are cleaning up anyway :)

Reply:
But if you have a gut feeling that this should stay - that is also fine.